### PR TITLE
feat: update slack docs

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

New Slack onboarding docs.

### Why was this change needed?

So the docs will be more informative in addition to the new onboarding flow.

### CODE REVIEW

File: docs/docs/channels/chat/slack.md [hunk 1]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
-<!-- markdownlint-disable MD029 -->
+## Slack Application Configuration
 
-7. You are all set up and ready to send your first chat message via our `@novu/node` package or directly using the REST API.
-<!-- markdownlint-enable MD029 -->
+1. Go to OAuth & Permissions on Slack's Developer Dashboard and add your REDIRECT_URL in Redirect URLs.
+   - If you use self-hosted solution add the api endpoint you created on [step 1](/channels/chat/slack#self-hosted).
+   - If you use Novu hosted solution add https:<span/>//api.novu.co/v1/subscribers/.
+2. Go to Incoming Webhooks from the left menu and Activate Incoming Webhooks.
+3. Go to Manage Distribution and at the bottom of the page make sure to tick Remove Hard Coded Information and Activate Public Distribution.
```
Comment (unicodeveloper) [inline docs/docs/channels/chat/slack.md:73]: => Go to Manage Distribution and at the bottom of the page, tick Remove Hard Coded Information and Activate Public Distribution.

File: docs/docs/channels/chat/slack.md [hunk 2]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
-:::
+</details>
 
-<!-- markdownlint-disable MD029 -->
+## Slack Application Configuration
 
-7. You are all set up and ready to send your first chat message via our `@novu/node` package or directly using the REST API.
-<!-- markdownlint-enable MD029 -->
+1. Go to OAuth & Permissions on Slack's Developer Dashboard and add your REDIRECT_URL in Redirect URLs.
+   - If you use self-hosted solution add the api endpoint you created on [step 1](/channels/chat/slack#self-hosted).
```
Comment (unicodeveloper) [inline docs/docs/channels/chat/slack.md:70]: => If you use a self-hosted solution, add the API endpoint...

File: docs/docs/channels/chat/slack.md [hunk 3]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
-import { Novu, ChatProviderIdEnum } from '@novu/node';
+   const body = req.body; // From your HTTPS listener
+   await novu.subscribers.setCredentials('subscriberId', ChatProviderIdEnum.Slack, {
+     webhookUrl: body.incoming_webhook.url,
+   });
+   ```
 
-const novu = new Novu(process.env.NOVU_API_KEY);
+   - subscriberId is a custom identifier used when identifying your users within the Novu platform.
+   - providerId is a unique provider identifier. We recommend using our ChatProviderIdEnum to specify the provider.
```
Comment (unicodeveloper) [inline docs/docs/channels/chat/slack.md:53]: We have a lot of non-JS/TS users, can we replace ChatProviderEnum here with the actual values too?

So something like:

`providerId is a unique provider identifier. We recommend using our ChatProviderIdEnum (if you're using Node), or any of the following values (Slack, ..., ...) to specify the provider.`

File: docs/docs/channels/chat/slack.md [hunk 4]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
+
+This step is optional, if you already have a Slack application you can reuse it.
 
 1. Go to Slack's developer dashboard [https://api.slack.com/apps](https://api.slack.com/apps).
 2. Create a new application.
-3. Go to `Incoming Webhooks` from the left menu and Activate Incoming Webhooks.
-4. (Optional) To test your integration with your workspace, you could generate a test webhook here.
 
-## Generate webhooks for your users
+## Generation of webhooks urls for your subscribers
```
Comment (unicodeveloper) [inline docs/docs/channels/chat/slack.md:16]: => Webhook Urls

File: docs/docs/channels/chat/slack.md [hunk 5]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
+2. Add the `Add to Slack` button or the shareable URL to your application in order to request permission of access (scope: incoming-webhook).
+   <br/>
+   Make sure to use the generated sharable URL that is located under Slack form in the <a href="https://web.novu.co/integrations" style={{textDecoration: "underline"}}>Integration Store</a>.
+
+### Manual Manage
+
+<details>
+<summary>Instructions on configuring https server</summary>
+
+Create a new endpoint on your server that will handle the following steps (you could you can use Request Bin for an easy HTTPS service for redirects):
```
Comment (unicodeveloper) [inline docs/docs/channels/chat/slack.md:32]: => you can use Request Bin for an easy HTTPS service for redirects

File: docs/docs/channels/chat/slack.md [hunk 6]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
+### Novu Managed (Recommended)
+
+Novu will manage the OAuth flow and store the credentials
+
+1. Configure your Slack application as mentioned <span style={{textDecoration: "underline"}}>[below](/channels/chat/slack#slack-application-configuration)</span>.
+2. Add the `Add to Slack` button or the shareable URL to your application in order to request permission of access (scope: incoming-webhook).
+   <br/>
+   Make sure to use the generated sharable URL that is located under Slack form in the <a href="https://web.novu.co/integrations" style={{textDecoration: "underline"}}>Integration Store</a>.
+
+### Manual Manage
```
Comment (p-fernandez) [inline docs/docs/channels/chat/slack.md:27]:
```suggestion
### Manual Management
```

File: docs/docs/channels/chat/slack.md [hunk 7]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
+
+<details>
+<summary>Instructions on configuring https server</summary>
+
+Create a new endpoint on your server that will handle the following steps (you can use Request Bin for an easy HTTPS service for redirects):
+
+1. Listen for redirect requests to your endpoint (REDIRECT_URL) after the user completes step 5 and grants permissions. Make sure to store the 'code' parameter from the request query as it will be needed later.
+2. Send a POST request to <https://slack.com/api/oauth.v2.access> with the following request body:
+   Use the "Client ID" and "Client Secret" from Slack's Developer Dashboard under "Basic Information". The request body should be in the format: { code: string, client_id: string, client_secret: string }.
+   Store the webhook URL from the response, which can be found under response.data.incoming_webhook.url.
```
Comment (p-fernandez) [inline docs/docs/channels/chat/slack.md:37]:
```suggestion
   Store the webhook URL from the response, which can be found under `response.data.incoming_webhook.url`.
```

File: docs/docs/channels/chat/slack.md [hunk 8]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
+   (read more on Slack's documentation here)
+3. When the incoming_webhook.url is obtained you need to save it on the relevant subscriber entity in Novu you can use the node sdk:
```
Comment (p-fernandez) [inline docs/docs/channels/chat/slack.md:39]:
```suggestion
3. When the `incoming_webhook.url` is obtained you need to save it on the relevant subscriber entity in Novu you can use the Node SDK:
```

File: docs/docs/channels/chat/slack.md [hunk 9]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
-```typescript
-import { Novu, ChatProviderIdEnum } from '@novu/node';
+   const body = req.body; // From your HTTPS listener
+   await novu.subscribers.setCredentials('subscriberId', ChatProviderIdEnum.Slack, {
+     webhookUrl: body.incoming_webhook.url,
+   });
+   ```
 
-const novu = new Novu(process.env.NOVU_API_KEY);
+   - subscriberId is a custom identifier used when identifying your users within the Novu platform.
```
Comment (p-fernandez) [inline docs/docs/channels/chat/slack.md:52]: I think we should ask an internal link to our documentation where we explain how to set up subscribers.

File: docs/docs/channels/chat/slack.md [hunk 10]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
+   - providerId is a unique provider identifier. We recommend using our ChatProviderIdEnum (if you're using Node), or any of the following values ('slack', 'discord', 'msteams', 'mattermost') to specify the provider.
```
Comment (p-fernandez) [inline docs/docs/channels/chat/slack.md:53]: Don't get why in the Slack documentation we mention other providers. I find it confusing.

File: docs/docs/channels/chat/slack.md [hunk 11]
```diff
@@ -6,42 +6,68 @@ This guide will walk you through the steps needed to obtain the `webhookUrl` tha
...
+   :::info
+   You need to set credentials for every subscriber because Slack generates a new Webhook URL on every new app install.
+   :::
 
-- `subscriberId` is a custom identifier used when identifying your users within the Novu platform.
-- `providerId` is a unique provider identifier. We recommend using our ChatProviderIdEnum to specify the provider.
-- The third parameter is the credentials object. In this case we use the `webhookUrl` property to specify the webhook URL generated in the previous step.
+4. Configure your Slack application as mentioned <span style={{textDecoration: "underline"}}>[below](/channels/chat/slack#slack-application-configuration)</span>.
+5. Add the `Add to Slack` button or the shareable URL to your application in order to request permission of access (scope: incoming-webhook).
+6. After the end-user finishes the authorization you will get the webhookUrl from the response of the OAuth under body.incoming_webhook.url, that you will use in step 3.
```
Comment (p-fernandez) [inline docs/docs/channels/chat/slack.md:62]:
```suggestion
6. After the end-user finishes the authorization you will get the webhookUrl from the response of the OAuth under `body.incoming_webhook.url`, that you will use in step 3.
```

**Review by p-fernandez — APPROVED** [review]
🌟

File: docs/docs/channels/chat/slack.md (hunk 10 above)
Comment (djabarovgeorge) [reply docs/docs/channels/chat/slack.md:53]: Yeah was not sure about it as well 😅

File: docs/docs/channels/chat/slack.md (hunk 9 above)
Comment (djabarovgeorge) [reply docs/docs/channels/chat/slack.md:52]: What do you mean? to add documentation that explains what subscriber is on our system?
(small note: in this case, it is the user's subscriber id.)

Comment (p-fernandez) [reply docs/docs/channels/chat/slack.md:52]: To add a link to this page: https://docs.novu.co/platform/subscribers

**Pull request merged**